---
title: "MICB425 Group 1 Project 2"
author: "Team 1: Albert Chang (), Alison Fong (33399149), Karen Lau (16524143), Yaqian
  Luo (59751503), Jessica Ngo (31837131), Peter (Kiet) Truong (36645133)"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output:
  html_document:
    df_print: paged
    toc: yes
    toc_float:
      collapsed: no
---

####Abstract  

####Introduction  

Oxygen minimum zones (OMZs) are regions in the ocean where dissolved oxygen concentrations fall below 20 $\mu$M (1). Because of temperature increases and other effects caused by global warming, OMZs are rising at a notable rate. Saanich Inlet is a seasonally anoxic fjord off the coast of British Columbia and a model ecosystem for studying the diversity and biochemical responses of microbial communities to the hypoxic environments commonly observed in OMZs (1, 2). In particular, Saanich Inlet has been used to model the metabolic coupling and symbiotic interactions that occur in OMZs (3).  The inlet undergoes recurring cycles of water column stratification and deep water renewal, making it a model ecosystem for studying microbial responses to changes in oceanic oxygenation levels (4). Increased levels of primary productivity in ocean surfaces during the spring season, along with the limited mixing which occurs between the basin and surface waters both result in the development of an anoxic body of water with increasing depth in the Inlet (2). These anoxic regions become highly populated with chemolithoautotrophs, eventually leading to a decrease in aerobically respiring organisms found deeper within these zones. Past studies have shown that these types of metabolic shifts usually lead to a loss of nitrogen along with the production of notable greenhouse gases such as methane (CH~4~) and nitrous oxide (N~2~O) (1). Some species have also been shown to engage in sulfide (H~2~S) oxidation and nitrate (NO~3~^-^) reduction pathways within these zones (5).  
The nitrogen cycle--the biogeochemical cycle by which nitrogenous compounds are interconverted between chemical forms for environmental circulation--consists mainly of nitrogen fixation, nitrification and denitrification, and has been catalyzed by microorganisms long before the appearance of humans (6). Even today, much of it is still dictated by the diverse microbial niches present in the surrounding environment. An example of this control can be seen in denitrification--the conversion of NO~3~^-^ to nitrogen gas (N~2~)--carried out by denitrifying bacteria. Denitrication is a highly important process as it prevents the accumulation of nitrogen compounds to toxic levels that could lead to eutrophication, and maintains the homeostasis of nitrogen distribution between soil and atmosphere. Having an atmospheric residence time of approximately 1 billion years, nitrogen gas (N~2~) is highly inert and is not accessible for the synthesis of proteins and nucleic acids, though this problem is easily solved by the conversion of N~2~ to NH~4~^+^ through nitrogen fixation (6).  
The aim of our project was to characterize the involvement of our gene of interest, narG, which is known for its role in the reduction of NO~3~^-^ to NO~2~^-^ in the denitrification cycle. Previous data has shown that (insert certain depths) exhibits traits of high narG activity: having low concentrations of (insert certain chemical) and (insert certain chemical), as seen in (insert figures) (adapted from certain paper). In this project, the analyzed data shows that at (insert a certain depth), there is high concentrations of (insert chemicals) and low concentrations of (certain chemicals), which is concurrent with the previous study. This is relevant because (insert certain reason). Recent literature has highlighted that microbial taxa do not necessarily integrate whole elemental cycles into their individual metabolic pathways, but often divide these reactions among the community so that consortia of taxa must rely on each other to fulfill their metabolic requirements (2).  
  



####Methods  

####Results  

#####Environment setup and Data Cleaning  
```{r echo=FALSE}
library(tidyverse)
library(cowplot)
```

Loading, parsing (initial data contains data for all genes - we are only interested in narG), renaming data.  We only need taxonomy, abundance, and query data.
```{r message=FALSE, echo=FALSE}
narG.DNA.10m = read_tsv("narG/marker_contig/DNA10m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%
    select(Tax.DNA.10 = Confident_Taxonomy, Abund.DNA.10 = Abundance, Query)

narG.RNA.10m = read_tsv("narG/marker_contig/RNA10m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>% 
    select(Tax.RNA.10 = Confident_Taxonomy, Abund.RNA.10 = Abundance, Query)

narG.DNA.100m = read_tsv("narG/marker_contig/DNA100m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>% 
    select(Tax.DNA.100 = Confident_Taxonomy, Abund.DNA.100 = Abundance, Query)

narG.RNA.100m = read_tsv("narG/marker_contig/RNA100m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%  
    select(Tax.RNA.100 = Confident_Taxonomy, Abund.RNA.100 = Abundance, Query)

narG.DNA.120m = read_tsv("narG/marker_contig/DNA120m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%  
    select(Tax.DNA.120 = Confident_Taxonomy, Abund.DNA.120 = Abundance, Query)

narG.RNA.120m = read_tsv("narG/marker_contig/RNA120m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%  
    select(Tax.RNA.120 = Confident_Taxonomy, Abund.RNA.120 = Abundance, Query)

narG.DNA.135m = read_tsv("narG/marker_contig/DNA135m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%  
    select(Tax.DNA.135 = Confident_Taxonomy, Abund.DNA.135 = Abundance, Query)

narG.RNA.135m = read_tsv("narG/marker_contig/RNA135m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%  
    select(Tax.RNA.135 = Confident_Taxonomy, Abund.RNA.135 = Abundance, Query)

narG.DNA.150m = read_tsv("narG/marker_contig/DNA150m_marker_contig_map.tsv") %>%
    filter(Marker=="narG") %>%  
    select(Tax.DNA.150 = Confident_Taxonomy, Abund.DNA.150 = Abundance, Query)

narG.RNA.150m = read_tsv("narG/marker_contig/RNA150m_marker_contig_map.tsv") %>% 
    filter(Marker=="narG") %>% 
    select(Tax.RNA.150 = Confident_Taxonomy, Abund.RNA.150 = Abundance, Query)

narG.DNA.165m = read_tsv("narG/marker_contig/DNA165m_marker_contig_map.tsv") %>% 
    filter(Marker=="narG") %>% 
    select(Tax.DNA.165 = Confident_Taxonomy, Abund.DNA.165 = Abundance, Query)

narG.RNA.165m = read_tsv("narG/marker_contig/RNA165m_marker_contig_map.tsv") %>% 
    filter(Marker=="narG") %>% 
    select(Tax.RNA.165 = Confident_Taxonomy, Abund.RNA.165 = Abundance, Query)

narG.DNA.200m = read_tsv("narG/marker_contig/DNA200m_marker_contig_map.tsv") %>% 
    filter(Marker=="narG") %>% 
    select(Tax.DNA.200 = Confident_Taxonomy, Abund.DNA.200 = Abundance, Query)

narG.RNA.200m = read_tsv("narG/marker_contig/RNA200m_marker_contig_map.tsv") %>% 
    filter(Marker=="narG") %>% 
    select(Tax.RNA.200 = Confident_Taxonomy, Abund.RNA.200 = Abundance, Query)
```

Data manipulation into a single data frame
```{r, echo=FALSE}
narG.all = narG.DNA.10m %>% 
# Combine the data frames will full_join to keep all the data
  full_join(narG.RNA.10m, by = "Query") %>%
  full_join(narG.DNA.100m, by = "Query") %>% 
  full_join(narG.RNA.100m, by = "Query") %>%
  full_join(narG.DNA.120m, by = "Query") %>% 
  full_join(narG.RNA.120m, by = "Query") %>%
  full_join(narG.DNA.135m, by = "Query") %>% 
  full_join(narG.RNA.135m, by = "Query") %>%
  full_join(narG.DNA.150m, by = "Query") %>% 
  full_join(narG.RNA.150m, by = "Query") %>%
  full_join(narG.DNA.165m, by = "Query") %>% 
  full_join(narG.RNA.165m, by = "Query") %>% 
  full_join(narG.DNA.200m, by = "Query") %>% 
  full_join(narG.RNA.200m, by = "Query") %>% 
# Create a taxonomy variable aggregating all taxonomy columns so as to fill in any NAs that might occur. !is.na means "is not NA", so the following says that the Taxonomy data should be taken from Tax.RNA.10 if that is not NA, or else take it from Tax.DNA.10 if that is not NA, or else Tax.RNA.200, etc. until if all are NA, give Taxonomy of "unclassified"
  mutate(Taxonomy = ifelse(!is.na(Tax.RNA.10), Tax.RNA.10,
                    ifelse(!is.na(Tax.DNA.10), Tax.DNA.10,
                    ifelse(!is.na(Tax.RNA.100), Tax.RNA.100,
                    ifelse(!is.na(Tax.DNA.100), Tax.DNA.100,
                    ifelse(!is.na(Tax.RNA.120), Tax.RNA.120,
                    ifelse(!is.na(Tax.DNA.120), Tax.DNA.120,
                    ifelse(!is.na(Tax.RNA.135), Tax.RNA.135,
                    ifelse(!is.na(Tax.DNA.135), Tax.DNA.135,
                    ifelse(!is.na(Tax.RNA.150), Tax.RNA.150,
                    ifelse(!is.na(Tax.DNA.150), Tax.DNA.150,
                    ifelse(!is.na(Tax.RNA.165), Tax.RNA.165,
                    ifelse(!is.na(Tax.DNA.165), Tax.DNA.165,
                    ifelse(!is.na(Tax.RNA.200), Tax.RNA.200,
                    ifelse(!is.na(Tax.DNA.200), Tax.DNA.200,
                           "unclassified"))))))))))))))) %>% 
# Get rid of the old Tax variables
  select(-starts_with("Tax.")) %>% 
# Gather all the abundance data into 1 column 
  gather("Key", "Abundance", starts_with("Abund")) %>% 
# Turn the Key into Depth and RNA/DNA variables. We can easily do this because we specifically named these variables with period separation when we loaded in the original data
  separate(Key, c("Key","Type","Depth_m"), by = ".") %>% 
# Remove Key variable now that it only contains "Abund". This also serves to reorder the columns so that the very long Query is at the end.
  select(Depth_m, Type, Abundance, Taxonomy, Query) %>% 
# Make sure R knows depth is numerical since it came from a character variable
  mutate(Depth_m = as.numeric(Depth_m)) %>% 
# Separate Taxonomy into columns so we can plot at different levels
  separate(Taxonomy, into = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep="; ")
```

Warning above ignored, because not all queries could be classified down to species level, so these cells are filled as NA, which is fine.

##### Final data table
Table 1. Final data of narG gene 
```{r, echo=FALSE}
narG.all
```

##### 1. How does the DNA abundance of narG gene differ with depth?
```{r, echo=FALSE}
narG.all %>% 
# Filter the DNA data
  filter(Type == "DNA") %>% 
# Change NAs to "unclassified" at the level you want to plot. Here we will do Genus
  mutate(Genus = ifelse(is.na(Phylum), "unclassified", Phylum)) %>% 
  group_by(Depth_m) %>%
  mutate(abund_sum=sum(Abundance, na.rm=TRUE))%>%
  group_by(Depth_m)%>%
  summarize(abund_sum=mean(abund_sum))%>%

  
ggplot(aes(x = "narG", y = Depth_m)) +
# Use the size aesthetic to show abundance
  geom_point(aes(size = abund_sum)) +
# Reverse the why axis so depth increases going down
  scale_y_reverse(lim=c(200,0)) +
  labs(y="Depth (m)")+
  theme_classic() +
  scale_size_continuous(name="Abundance",breaks = c(200,400,600,800)) +
  geom_text(aes(label=abund_sum, hjust=1.5))+
  labs(caption = "Figure 1. Abundance of the narG gene (DNA) at different depths",
       x = "") +
  theme(plot.caption = element_text(hjust=0.5, size=rel(1.0)))
```
 

##### 2. What taxa are responsible for narG? Are they the same with depth? With DNA versus RNA?
```{r, echo=FALSE}
# DNA and RNA abundance with depth
DNA.abund.depth=narG.all %>% 
  filter(Type == "DNA") %>% 
  mutate(Genus = ifelse(is.na(Phylum), "unclassified", Phylum)) %>% 
  group_by(Depth_m) %>%
  mutate(abund_sum=sum(Abundance, na.rm=TRUE))%>%
  group_by(Depth_m)%>%
  summarize(DNA=mean(abund_sum))

RNA.abund.depth=narG.all %>% 
  filter(Type == "RNA") %>% 
  mutate(Genus = ifelse(is.na(Phylum), "unclassified", Phylum)) %>% 
  group_by(Depth_m) %>%
  mutate(abund_sum=sum(Abundance, na.rm=TRUE))%>%
  group_by(Depth_m)%>%
  summarize(RNA=mean(abund_sum))

left_join(DNA.abund.depth,RNA.abund.depth,by="Depth_m")%>%
  gather(Type,abund_sum,DNA:RNA)%>%

# Show both RNA and DNA using an x variable  
ggplot(aes(x = Type, y = Depth_m)) +
  geom_point(aes(size = abund_sum)) +
  labs(y="Depth (m)")+
  scale_y_reverse(lim=c(200,0)) +
  theme_classic()+
  scale_size_continuous(name="Abundance",breaks = c(500,1000,1500,2000,3000)) +
  geom_text(mapping=aes(label=abund_sum),hjust=1.2)+
  labs(caption = "Figure 2. Abundance of the narG gene (DNA vs. RNA) at different depths") +
  theme(plot.caption = element_text(hjust=0.5, size=rel(1.0)))
```

##### 3. What taxa are responsible for narG gene? Are they the same with depth? With DNA versus RNA?
```{r fig.width=8, echo=FALSE}
narG.all %>% 
# Change NAs to "unclassified" at the level you want to plot
  mutate(Phylum = ifelse(is.na(Phylum), "unclassified Bacteria", Phylum)) %>% 
  
ggplot(aes(x = Phylum, y = Depth_m)) +
# Use an ifelse statement to make 0 values into NAs so that they don't show up on the plot
# Use position_dodge to keep points from overlapping
  geom_point(aes(size = ifelse(Abundance == 0, NA, Abundance), color = Type), position = position_dodge(0.5)) +
  scale_y_reverse(lim=c(200,0)) +
  labs(y="Depth (m)")+
  theme_classic() +
# Rename legend
  scale_size_continuous(name = "Abundance") +  
  theme(axis.text.x = element_text(angle = 45, hjust =1, size=rel(1.2)))+
  labs(caption = "Figure 3. Abundance of phyla with narG (DNA vs. RNA) at different depths") +
  theme(plot.caption = element_text(hjust=0.5, size=rel(1.1)))
 
```


##### 4. How does the abundance of narG gene relate to nitrogen species in Saanich?

Loading in data from project 1 and pulling out the geochemical metadata.
```{r fig.width=7, fig.height=7, echo=FALSE}
load("mothur_phyloseq.RData")

metadata = data.frame(mothur@sam_data)

# DNA/RNA abundance with depth
plot1 = narG.all %>% 
# Change NAs to "unclassified" at the level you want to plot
  mutate(Phylum = ifelse(is.na(Phylum), "unclassified", Phylum)) %>% 
  
ggplot(aes(x = Phylum, y = Depth_m)) +
  geom_point(aes(size = ifelse(Abundance == 0, NA, Abundance), color = Type), position = position_dodge(0.5)) +
  scale_y_reverse(lim=c(200,0)) +
  labs(y = "Depth (m)") +
  theme_classic() +
  scale_size_continuous(name = "Abundance")+ 
  theme(axis.text.x = element_text(angle = 45, hjust =1))

#Then, we make our plots for NO2, "plot2", and NO3, "plot3"

plot2 = metadata %>% 
# Order the data by depth  
  arrange(Depth_m) %>% 
  
ggplot(aes(x = NO2_uM, y = Depth_m)) +
    geom_point() +
    geom_path(aes(group = 1)) +
    scale_y_reverse(lim=c(200,0)) +
  theme_classic() +
  labs(y = "Depth (m)",
       x = "NO2 (uM)")
plot3 = metadata %>% 
# Order the data by depth  
  arrange(Depth_m) %>% 
  
ggplot(aes(x = NO3_uM, y = Depth_m)) +
    geom_point() +
    geom_path(aes(group = 1)) +
    scale_y_reverse(lim=c(200,0)) +
  theme_classic() +
  labs(y = "Depth (m)",
       x = "NO3 (uM)")

plot4 = metadata %>% 
# Order the data by depth  
  arrange(Depth_m) %>% 
  
ggplot(aes(x = NH4_uM, y = Depth_m)) +
    geom_point() +
    geom_path(aes(group = 1)) +
    scale_y_reverse(lim=c(200,0)) +
  theme_classic() +
  labs(y = "Depth (m)",
       x = "NH4 (uM)")

plot5 = metadata %>% 
# Order the data by depth  
  arrange(Depth_m) %>% 
  
ggplot(aes(x = N2O_nM, y = Depth_m)) +
    geom_point() +
    geom_path(aes(group = 1)) +
    scale_y_reverse(lim=c(200,0)) +
  theme_classic() +
  labs(y = "Depth (m)",
       x = "N2O (nM)")

plot_grid(plot4, plot5, plot2, plot3, labels=c("A", "B", "C", "D"), axis="l", nrow=2, ncol=2, rel_widths=c(1/2, 1/2, 1/2,1/2))+
  labs(caption = "Figure 4. Nitrogen species in Saanich across depth. 
A) Ammonium (NH4) concentration across depth   B) Nitrous oxide (N2O) concentration across depth 
C) Nitrite (NO2) concentration across depth     D) Nitrate (NO3) concentration across depth") +
  theme(plot.caption = element_text(hjust=0.5, size=rel(0.75)))
```

####Discussion

```{r fig.width=5, fig.height=5}
# Oxygen with depth
metadata %>% 
# Order the data by depth  
  arrange(Depth_m) %>% 
  
ggplot(aes(x = O2_uM, y = Depth_m)) +
    geom_point() +
    geom_path(aes(group = 1)) +
    scale_y_reverse(lim=c(200,0)) +
  theme_classic() +
  labs(y = "Depth (m)",
       x = "O2 (uM)")+
  labs(caption="Figure 5. Oxygen in Saanich across depth")+
  theme(plot.caption = element_text(hjust=0.5,size=rel(1.0)))
```

####References  

1. Walsh DA, Zaikova E, Howes CG, Song YC, Wright JJ, Tringe SG, Hallam SJ. 2009. Metagenome of a versatile chemolithoautotroph from expanding oceanic dead zones. Science 326(5952): 578-582.  
2. Torres-Beltrán M, Hawley AK, Capelle D, Zaikova E, Walsh DA., Mueller A, Finke J. 2017. A compendium of geochemical information from the Saanich Inlet water column. Nature scientific data 4(170159).  
3. Hawley AK, Torres-Beltrán M, Zaikova E, Walsh DA, Mueller A, Scofield M, Kheirandish S, Payne C, Pakhomova L, Bhatia M, Shevchek O, Gies EA, Fairley D, Malfatii SA, Norbeck AD, Brewer HM, Pasa-Tolic, L, del Rio TG, Suttle CA, Trige S, Hallam SJ. Data Descriptor: A compendium of multi-omic sequence information from the Saanich Inlet water column. Nature scientific data 4(170160).  
4. Hallam SJ, Torres-Beltrán M, Hawley AK. Comment: Monitoring microbial responses to ocean deoxygenation in a model oxygen minimum zone. Nature scientific data 4(170158).  
5. National Research Council. 2007. The new science of metagenomics: revealing the secrets of our microbial planet. National Academies Press, Washington, DC.  
6. Falkowski PG, Fenchel T, Delong EF. 2008. The microbial engines that drive Earth's biogeochemical cycles. Science 320(5879): 1034-1039.
